### PR TITLE
Issue #12071: Don't submit forms when typing Tab on an input.

### LINKED ISSUE #12071: using tab for username/pass entry causes submit
Decided to load Hacker News as first site given I saw Servo there. Went to login page. Typed username. Hit tab to go to password field. Automatically submitted the form with bad login. I used mouse to highlight and delete both field to see if it was reproducible. Typed username, hit tab, and form submitted. 

Using Servo on Ubuntu 16.04 (bare metal) with -c option recommended in "can't launch" issue since it otherwise wouldn't run. Did nothing else different.

### PR DESCRIPTION
**Pull request by simartin**
## Fixes https://github.com/servo/servo/issues/12071

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12071

- [X] There are tests for these changes

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12365)

### CODE REVIEW

File: components/script/dom/htmlinputelement.rs [hunk 1]
```diff
@@ -1046,10 +1047,17 @@ impl VirtualMethods for HTMLInputElement {
...
-                                                     keyevent.AltKey(),
-                                                     keyevent.MetaKey());
+                            if let Some(key) = keyevent.get_key() {
+                                match key {
+                                    Key::Enter | Key::KpEnter =>
+                                        self.implicit_submission(keyevent.CtrlKey(),
+                                                                 keyevent.ShiftKey(),
+                                                                 keyevent.AltKey(),
+                                                                 keyevent.MetaKey()),
+                                    // Issue #12071
```
Comment (notriddle) [inline components/script/dom/htmlinputelement.rs]: Could you include an actual synopsis? Something like

    // TODO(12071): Implement tab focus switching

Comment (simartin) [inline components/script/dom/htmlinputelement.rs]: Thanks for the suggestion and the issue # for keyboard navigation. Done in the updated patch

File: tests/wpt/mozilla/tests/mozilla/form_tab_keyevent.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			document.addEventListener("DOMContentLoaded", function() {
+			  test.step(function() {
+			  	var input = document.getElementById('input1');
+			  	var evt = new KeyboardEvent("keydown", {key : "Tab"});
+			  	input.dispatchEvent(evt);
+			  });
+			});
+			test.step_timeout(function() { assert_false(form_submitted, "Form should not be submitted"); test.done(); }, 500);
+		</script>
+	</body>
```
Comment (KiChjang) [inline tests/wpt/mozilla/tests/mozilla/form_tab_keyevent.html]: Woah, there are tabs in this test file. Doubleplusungood. Please change them into spaces (either 2 or 4 spaces per indent are fine).

**Comment by notriddle** [comment]
Why didn't tidy catch it?

**Comment by simartin** [comment]
*[tests/wpt/mozilla/tests/mozilla/form_tab_keyevent.html, line 26 \[r2\]](https://reviewable.io:443/reviews/servo/servo/12365#-KMFcOf_Zd737WHhmhQi-r2-26:-KMW51qHa0Y1NNnWGHIG:1161068296) ([raw file](https://github.com/servo/servo/blob/9547633e9532083aa46017cfbd889b2dbddc6ec8/tests/wpt/mozilla/tests/mozilla/form_tab_keyevent.html#L26)):*
<details><summary><i>Previously, KiChjang (Keith Yeung) wrote…</i></summary>
> Woah, there are tabs in this test file. Doubleplusungood. Please change them into spaces (either 2 or 4 spaces per indent are fine).
</details>
Patch updated accordingly. Thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/12365)*

**Comment by simartin** [comment]
tidy did not catch it because it does not consider *.html files. I've opened https://github.com/servo/servo/issues/12421 to fix this.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/12365#-:-KMWGHvgu8qyZpgoiW84:-1694431130)*

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Pull request merged**